### PR TITLE
[KeyVault] Bump api-version to 2023-02

### LINKED ISSUE #2165: Keyvault latest version 1.1.0 does not support secrets.
### Link to sample

_No response_

### Library name and version

Azure.ResourceManager.Keyvault

### Language of the Sample

- [X] C#/.NET
- [ ] Java
- [ ] JavaScript/TypedScript
- [ ] Python
- [ ] Golang
- [ ] Other - Please specify in Issue details field

### Sample Issue Type

- [X] Sample not working
- [X] Sample missing
- [ ] Do not understand sample

### Issue details

We are migrating from **Microsoft.Azure.Management.KeyVault** SDK to the new **Azure.ResourceManager.KeyVault** SDK (latest 1.1.0)

1. We noticed that the latest version did not contain Secret resource or any capability to support secrets. No example found for the same.
2. We downgraded the version to 1.0.0.-beta8. Even though it supports SecretResource, we are unable to fetch any secrets in the code. The value remains null.

Please provide examples or guidance to fetch Secret resource.

### Expected behavior

_No response_

### Actual behavior

_No response_

### Reproduction Steps

_No response_

### Environment

development

### PR DESCRIPTION
**Pull request by dvbb**
Fixes https://github.com/Azure/azure-rest-api-specs-examples/issues/2165

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by dvbb** [comment]
Linked issue: https://github.com/Azure/azure-rest-api-specs-examples/issues/2165

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 1]
```diff
@@ -944,10 +1070,61 @@ public enum ManagedHsmSkuName
...
         CustomB32 = 1,
+        CustomB6 = 2,
     }
     public partial class ManagedHsmVirtualNetworkRule
     {
         public ManagedHsmVirtualNetworkRule(Azure.Core.ResourceIdentifier subnetId) { }
         public Azure.Core.ResourceIdentifier SubnetId { get { throw null; } set { } }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct Reason : System.IEquatable<Azure.ResourceManager.KeyVault.Models.Reason>
```
Comment (ArcturusZhang) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:1081]: Could we rename this to `ManagedHsmNameUnavailableReason`?

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 2]
```diff
@@ -284,6 +326,47 @@ public enum AccessPolicyUpdateKind
...
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.ResourceManager.KeyVault.Models.ActivationStatus left, Azure.ResourceManager.KeyVault.Models.ActivationStatus right) { throw null; }
+        public static implicit operator Azure.ResourceManager.KeyVault.Models.ActivationStatus (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.KeyVault.Models.ActivationStatus left, Azure.ResourceManager.KeyVault.Models.ActivationStatus right) { throw null; }
+        public override string ToString() { throw null; }
+    }
+    public partial class Attributes
```
Comment (ArcturusZhang) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:349]: this name is too generic, please find the type or property that are using this type and prepend a proper prefix to this type

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 3]
```diff
@@ -311,6 +394,28 @@ public partial class DeletedManagedHsmProperties
         public System.Collections.Generic.IReadOnlyDictionary<string, string> Tags { get { throw null; } }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct GeoReplicationRegionProvisioningState : System.IEquatable<Azure.ResourceManager.KeyVault.Models.GeoReplicationRegionProvisioningState>
```
Comment (ArcturusZhang) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:397]: Could we rename this to `ManagedHsmGeoReplicatedRegionProvisioningState`?

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 4]
```diff
@@ -797,6 +913,8 @@ public partial class ManagedHsmNetworkRuleSet
     public partial class ManagedHsmPrivateEndpointConnectionItemData
     {
         internal ManagedHsmPrivateEndpointConnectionItemData() { }
+        public Azure.ETag? ETag { get { throw null; } }
+        public string Id { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:917]: Do we know if this is an arm-id? If so, we should change the format here

Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs]: According to the [example](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2023-02-01/examples/ManagedHsm_putPrivateEndpointConnection.json#L20), it is an `arm-id`

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs (hunk 2 above)
Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs]: rename to `SecretBaseAttributes`

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 5]
```diff
@@ -284,6 +326,47 @@ public enum AccessPolicyUpdateKind
         Replace = 1,
         Remove = 2,
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ActivationStatus : System.IEquatable<Azure.ResourceManager.KeyVault.Models.ActivationStatus>
```
[the remark this answers is not in the document]
Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs]: rename to `ManagedHSMSecurityDomainActivationStatus`

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 6]
```diff
@@ -917,6 +1037,12 @@ public partial class ManagedHsmProperties
         public static bool operator !=(Azure.ResourceManager.KeyVault.Models.ManagedHsmPublicNetworkAccess left, Azure.ResourceManager.KeyVault.Models.ManagedHsmPublicNetworkAccess right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class ManagedHSMSecurityDomainProperties
```
[the remark this answers is not in the document]
Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:1049]:
```
  Mhsm: ManagedHsm
```
We already have a rule in `rename-rules`

File: sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Tests/EventHubNamespaceTests.cs [hunk 7]
```diff
@@ -376,6 +376,7 @@ public async Task PremiumNamespaceCreateOrUpdate()
 
         [Test]
         [RecordedTest]
+        [Ignore("Depende on KeyVault which will block the pipeline to release new SDK, disable this case temporary")]
```
Comment (ArthurMa1978) [inline sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Tests/EventHubNamespaceTests.cs:379]: Please have another pr to decouple with the KV dependency first.

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 8]
```diff
@@ -268,13 +274,49 @@ public partial class ManagedHsmResource : Azure.ResourceManager.ArmResource
...
+        public virtual Azure.AsyncPageable<Azure.ResourceManager.KeyVault.SecretResource> GetAllAsync(int? top = default(int?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.KeyVault.SecretResource>> GetAsync(string secretName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        System.Collections.Generic.IAsyncEnumerator<Azure.ResourceManager.KeyVault.SecretResource> System.Collections.Generic.IAsyncEnumerable<Azure.ResourceManager.KeyVault.SecretResource>.GetAsyncEnumerator(System.Threading.CancellationToken cancellationToken) { throw null; }
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.KeyVault.SecretResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.KeyVault.SecretResource>.GetEnumerator() { throw null; }
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class SecretData : Azure.ResourceManager.Models.ResourceData
```
Comment (ArthurMa1978) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:301]: Secret is a single word

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs [hunk 9]
```diff
@@ -745,11 +796,65 @@ public enum ManagedHsmCreateMode
         Default = 0,
         Recover = 1,
     }
+    public partial class ManagedHsmGeoReplicatedRegion
+    {
+        public ManagedHsmGeoReplicatedRegion() { }
+        public bool? IsPrimary { get { throw null; } set { } }
+        public string Name { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:803]: Is this region name? and format is `azure-location`?

File: sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Tests/EventHubNamespaceTests.cs (hunk 7 above)
Comment (dvbb) [reply sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Tests/EventHubNamespaceTests.cs:379]: I will raise another PR to re-enable these cases after this PR merged

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs (hunk 8 above)
Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:313]: rename `Secret` => `KeyVaultSecret`

File: sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs (hunk 9 above)
Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/api/Azure.ResourceManager.KeyVault.netstandard2.0.cs:810]: Difficult to figure out if this is `azure-location`, decided not to change it in sync meeting

File: sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Tests/EventHubNamespaceTests.cs (hunk 7 above)
Comment (dvbb) [reply sdk/eventhub/Azure.ResourceManager.EventHubs/tests/Tests/EventHubNamespaceTests.cs:379]: Here is the remove PR https://github.com/Azure/azure-sdk-for-net/pull/35989

**Review by ArthurMa1978 — APPROVED** [review]

**Comment by ArthurMa1978** [comment]
Don't merge until https://github.com/Azure/azure-sdk-for-net/pull/35989 got merged, then remove all changes that not belong to the KeyVault sdk

**Pull request merged**